### PR TITLE
Rewrite anonym function "syncTree" for compiler and debugging reasons.

### PR DESCRIPTION
**Pull request by MarkusBordihn**
See #448 for more details.

### CODE REVIEW

File: core/toolbox.js [hunk 1]
```diff
@@ -265,92 +265,99 @@ Blockly.Toolbox.prototype.position = function() {
...
+
+  // Fire a resize event since the toolbox may have changed width and height.
+  Blockly.resizeSvgContents(this.workspace_);
+};
+
+
+/**
+ * Sync trees of the toolbox.
+ * @param {Node} treeIn DOM tree of blocks, or null.
+ * @param {Blockly.Toolbox.TreeControl} treeOut
```
Comment (rachel-fenichel) [inline core/toolbox.js]: Now that this isn't anonymous you should be able to get rid of the treeOut parameter and just used this.tree_ in the function.

**Comment by rachel-fenichel** [comment]
This diff is a bit hard to read--can you summarize the change you made?

It looks like you've moved syncTree to be defined on the toolbox prototype.  Anything else?

File: core/toolbox.js (hunk 1 above)
Comment (MarkusBordihn) [inline core/toolbox.js]: I have leave this, in the case that its needed later.

**Comment by MarkusBordihn** [comment]
Yes, in general I have moved syncTree to an prototype and adjust the parameters and removed unneeded var declarations. I have not changed the logic of syncTree itself.

File: core/toolbox.js [hunk 2]
```diff
@@ -265,92 +265,99 @@ Blockly.Toolbox.prototype.position = function() {
...
+
+  if (this.tree_.blocks.length) {
+    throw 'Toolbox cannot have both blocks and categories in the root level.';
+  }
+
+  // Fire a resize event since the toolbox may have changed width and height.
+  Blockly.resizeSvgContents(this.workspace_);
+};
+
+
```
Comment (rachel-fenichel) [inline core/toolbox.js]: One newline between functions, here and below.

**Comment by rachel-fenichel** [comment]
One comment then LGTM.

**Comment by MarkusBordihn** [comment]
Done.
